playground/src/Editor/SideBar.tsx

```diff
@@ -0,0 +1,61 @@
+import { FileIcon, SettingsIcon } from "./Icons";
+import { type ReactNode } from "react";
+
+type Tool = "Settings" | "Source";
+
+type SideBarProps = {
+  selected: Tool;
+  onSelectTool(tool: Tool): void;
+};
```

Comment:
I always just inline single-use definitions like this, e.g.:

```typescript
export default function SideBar({ selected, onSelectTool }: { selected: Tool, ... } { ... }
```

But it's entirely subjective :)

Reply:
I can't do that. :stuck_out_tongue: 

I find it extremely hard to read, always mess up the syntax when writing the types, and you need to refactor them out once you start adding new props.